server/game/core/ability/abilityTargets/PlayerTargetResolver.ts

```diff
@@ -0,0 +1,83 @@
+import { TargetResolver } from './TargetResolver';
+import { IPlayerTargetResolver } from '../../../TargetInterfaces';
+import { AbilityContext } from '../AbilityContext';
+import type Player from '../../Player';
+import PlayerOrCardAbility from '../PlayerOrCardAbility';
+import { Stage } from '../../Constants';
+
+// This currently assumes that every player will always be a legal target for any effect it's given.
+// TODO: Make a PlayerSelector class(see the use of property "selector" in CardTargetResolver) to help determine target legality. Use it to replace placeholder override functions below.
```

Comment:
I'm not 100% opposed to checking the class in like this, since it is providing some useful functionality and some of the missing pieces (specifically the legality checks) are not relevant given the current cards in the game.

My concern however is that people will try to use this in the future thinking that it has the same level of functionality as `CardTargetSelector` (understandably), which will cause a lot of frustration / debugging effort while they figure out that's not the case.

Unless you're planning to address these in the very near future (i.e. in your next PR or two), I think there are a few extra things we should do in this PR to maximize the utility of this and make it less likely that people will get tripped up:

- Add a similar TODO comment on IPlayerTargetResolver explaining that the functionality is incomplete, and one on `AbilityRestriction` in Constants.ts explaining that player restrictions are not currently respected by PlayerTargetResolver
- I think we should add enough functionality to be able to support existing cases like Yoda - i.e., the ability to also select "both" and "none." That seems like such an obvious feature that I doubt there's any amount of documenting we could do that would cause people to not expect it to be there. It should be very easy to add flags for those via `IPlayerTargetProperties` though, and then we can add it to Yoda in this PR to document the functionality.